system/jlib/jutil.cpp

```diff
@@ -2663,8 +2662,79 @@ StringBuffer &getFileAccessUrl(StringBuffer &out)
     return out;
 }
 
+
+#ifdef _CONTAINERIZED
+static bool getDefaultPlane(StringBuffer &ret, const char * componentOption, const char * globalOption)
+{
+    // If the plane is specified for the component, then use that
+    if (getComponentConfigSP()->getProp(componentOption, ret))
+        return true;
+
+    //Otherwise check what the default plane for data storage is configured to be
+    if (getGlobalConfigSP()->getProp(globalOption, ret))
+        return true;
+
+    return false;
+}
+
+static bool getDefaultPlaneDirectory(StringBuffer &ret, const char * componentOption, const char * globalOption)
+{
+    StringBuffer planeName;
+    if (!getDefaultPlane(planeName, componentOption, globalOption))
+        return false;
+
+    Owned<IPropertyTree> storagePlane = getStoragePlane(planeName);
+    return storagePlane->getProp("@prefix", ret);
+}
+#endif
+
 bool getConfigurationDirectory(const IPropertyTree *useTree, const char *category, const char *component, const char *instance, StringBuffer &dirout)
 {
+#ifdef _CONTAINERIZED
+    if (streq(category, "data"))
+    {
+        Owned<IPropertyTree> storagePlane = getStoragePlane(instance);
```

Comment:
There's code in Thor in thmastermain.cpp (820-831) which before this PR would always return null, and not call setBaseDirectory as consequence - so that code did nothing.
With this change it might return something if a plane matches the Thor name.
I think that code should be #ifdef 'd out in container.

Thor default to the default plane, if there are no CLUSTER/PLANE directives on an output.